crates/re_space_view_spatial/src/parts/assets.rs

```diff
@@ -29,51 +29,92 @@ impl Default for Asset3DPart {
 }
 
 impl Asset3DPart {
-    fn process_entity_view(
+    fn process_arch_view(
         &mut self,
         ctx: &mut ViewerContext<'_>,
         instances: &mut Vec<MeshInstance>,
-        ent_view: &EntityView<Mesh3D>,
+        arch_view: &ArchetypeView<Asset3D>,
         ent_path: &EntityPath,
         ent_context: &SpatialSceneEntityContext<'_>,
     ) -> Result<(), QueryError> {
-        let primary_row_id = ent_view.primary_row_id();
-        let visitor = |instance_key: InstanceKey,
-                       mesh: re_components::Mesh3D,
-                       _color: Option<Color>| {
-            let picking_instance_hash =
-                re_data_store::InstancePathHash::instance(ent_path, instance_key);
-
-            let outline_mask_ids = ent_context.highlight.index_outline_mask(instance_key);
-
-            let mesh = ctx.cache.entry(|c: &mut MeshCache| {
-                c.entry(
-                    &ent_path.to_string(),
-                    picking_instance_hash.versioned(primary_row_id),
-                    AnyMesh::Asset(&mesh),
-                    ctx.render_ctx,
-                )
-            });
-            if let Some(mesh) = mesh {
-                instances.extend(mesh.mesh_instances.iter().map(move |mesh_instance| {
-                    MeshInstance {
+        let transform = arch_view
+            .iter_raw_optional_component::<OutOfTreeTransform3D>()?
+            .and_then(|mut batch| batch.next());
+
+        let mesh = Asset3D {
+            data: arch_view
+                .iter_required_component::<Blob>()?
+                .next()
+                .ok_or_else(|| re_types::DeserializationError::MissingData {
+                    backtrace: re_types::_Backtrace::new_unresolved(),
+                })?,
+            media_type: arch_view
+                .iter_raw_optional_component::<MediaType>()?
+                .and_then(|mut batch| batch.next()),
+            // NOTE: Don't even try to cache the transform!
+            transform: None,
+        };
+
+        let primary_row_id = arch_view.primary_row_id();
+        let picking_instance_hash = re_data_store::InstancePathHash::entity_splat(ent_path);
+        let outline_mask_ids = ent_context.highlight.index_outline_mask(InstanceKey::SPLAT);
+
+        // TODO(#3232): this is subtly wrong, the key should actually be a hash of everything that got
+        // cached, which includes the media type...
+        let mesh = ctx.cache.entry(|c: &mut MeshCache| {
+            c.entry(
+                &ent_path.to_string(),
+                picking_instance_hash.versioned(primary_row_id),
```

Comment:
This comment would make more sense if I could see what the key is

```suggestion
        // TODO(#3232): this is subtly wrong, the key should actually be a hash of everything that got
        // cached, which includes the media type...
        let mesh = ctx.cache.entry(|c: &mut MeshCache| {
            let key = picking_instance_hash.versioned(primary_row_id);
            c.entry(
                &ent_path.to_string(),
                key,
```

Reply:
Can't we just add the mediatype to the key?

``` rs
let key = MeshKey {
    versioned_instance_path_hash: picking_instance_hash.versioned(primary_row_id),
    media_type: mesh.media_type
};
```

- seems like an easy thing to add

Reply:
Yeah that felt a bit too nasty to hash these two together... but I guess it's better than nothing while we wait for #3232